org.scala-ide.sdt.core.tests/src/org/scalaide/core/sbtbuilder/NonScalaSourceErrorMarkersTest.scala

```diff
@@ -0,0 +1,81 @@
+    val description = project.getDescription
+    description.setNatureIds(newNatures(description))
+    project.setDescription(description, null)
+  }
+
+  def removeScalaNature(project: IProject) = natures(project) { description =>
+    description.getNatureIds.filter { _ != SdtConstants.NatureId }
+  }
+
+  def soThen(assertThat: => Unit): Unit = assertThat
+
+  @Test def shouldFindJavaErrorMarkerTooWhenScalaNatureOfAisOffAndOn(): Unit = {
+    val Seq(prjA, prjB, prjC) = createProjects("A", "B", "C")
+
+    try {
+      prjB dependsOnAndExports prjA
+      prjC onlyDependsOn prjB
+
+      val Seq(packA, packB, packC) = Seq(prjA, prjB, prjC).map(createSourcePackage("test"))
+
+      val unitA = packA.createCompilationUnit("A.scala", "trait A", true, null)
+      val unitB = packB.createCompilationUnit("B.scala", "class B extends A", true, null)
+      val unitC = packC.createCompilationUnit("C.scala", "class C extends B", true, null)
+
+      // set stopOnBuild to true
+      val stopBuildOnErrors = SettingConverterUtil.convertNameToProperty(ScalaPluginSettings.stopBuildOnErrors.name)
+      IScalaPlugin().getPreferenceStore.setValue(stopBuildOnErrors, true)
+
+      // no errors
+      SDTTestUtils.workspace.build(IncrementalProjectBuilder.INCREMENTAL_BUILD, null)
+      val unitsToWatch = Seq(unitA, unitB, unitC)
+      val errors = SDTTestUtils.getErrorMessages(unitsToWatch: _*)
+      Assert.assertEquals("No build errors", Seq(), errors)
+
+      def whenRemoveAndAddScalaNatureToA(): Unit = {
+        removeScalaNature(prjA.underlying)
+        SDTTestUtils.workspace.build(IncrementalProjectBuilder.INCREMENTAL_BUILD, null)
+        addScalaNature(prjA.underlying)
+        SDTTestUtils.workspace.build(IncrementalProjectBuilder.INCREMENTAL_BUILD, null)
+      }
+      whenRemoveAndAddScalaNatureToA()
+
+      soThen {
+        Assert.assertEquals("One error in A", Seq((2, "Syntax error on tokens, delete these tokens")), SDTTestUtils.getErrorMessages(prjA.underlying))
+        Assert.assertEquals("One error in B", Seq((2, """Project: "B" in scope: "main" not built due to errors in dependent project(s): A. Root error(s): Syntax error on tokens, delete these tokens""")), SDTTestUtils.getErrorMessages(prjB.underlying))
+        Assert.assertEquals("One error in C", Seq((2, """Project: "C" in scope: "main" not built due to errors in dependent project(s): B, A. Root error(s): Project: "B" in scope: "main" not built due to errors in dependent project(s): A. Root error(s): Syntax error on tokens, delete these tokens;Syntax error on tokens, delete these tokens""")),
+          SDTTestUtils.getErrorMessages(prjC.underlying))
+      }
+    } finally {
+      deleteProjects(prjA, prjB, prjC)
+    }
+  }
+
+  @Ignore("@wpopielarski: not implemented because currently I see no way to fail scala installation choice")
+  @Test def shouldFindGlobalScalaErrorMarkerTooWhenNoEffectiveScalaInstallationFoundForA(): Unit = {}
```

Comment:
Actually I wanted to crash sbt or scala installation choice but I don't see now how to do this, so ok this test will be removed